src/Impostor.Server/Net/State/Game.Data.cs

```diff
@@ -348,6 +348,11 @@ private async ValueTask OnSpawnAsync(ClientPlayer sender, InnerNetObject netObj)
 
                 case InnerPlayerInfo playerInfo:
                 {
+                    if (!IsHostAuthoritive && await sender.Client.ReportCheatAsync(new CheatContext(nameof(GameDataTag.SpawnFlag)), CheatCategory.ProtocolExtension, "Spawning NetworkedPlayerInfo as vanilla host"))
```

Comment:
The rest of the Impostor code base does not combine `if(await sender.ReportCheatAsync(...))` with other if statements, because the encased logic block is usually also recycled for other things. Please split this if condition for consistency